src/Data/Auth/Providers/AuthContext.ts

```diff
@@ -0,0 +1,16 @@
+import { createContext } from "react";
```

Comment:
is "Props" the convention in ts for interfaces?
This also needs a comment that this is the interface.

Interfaces define the "contract" to the other code. So here it needs a clear documentation of what the semantics are of each of these functions

Reply:
Yes, "Props" it is a convention here. Added documentation